### DISCUSSION TITLE
Using TcpStream with deku and a library using deku with no_std support

### DISCUSSION THREAD

**Discussion by theoparis** [sharksforarms/deku#486] [Q&A] [answered]
Hi. I'm trying to find an easy way to implement a 9pfs protocol library. I tried to use Deku with from_reader but it errors because std::net::TcpStream doesn't implement Seek 😕 
It sounds like this might have worked before no_std_io was added - can this be fixed somehow?

```
error[E0277]: the trait bound `TcpStream: Seek` is not satisfied
   --> crates/p9_server/main.rs:7:54
    |
7   |     let (_rest, p9_message) = P9Message::from_reader((&mut stream, 0)).unwrap();
    |                               ---------------------- ^^^^^^^^^^^^^^^^ the trait `Seek` is not implemented for `TcpStream`
    |                               |
    |                               required by a bound introduced by this call
    |
note: required by a bound in `from_reader`
   --> /home/theo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/deku-0.18.1/src/lib.rs:459:41
    |
459 |     fn from_reader<R: no_std_io::Read + no_std_io::Seek>(
    |                                         ^^^^^^^^^^^^^^^ required by this bound in `DekuContainerRead::from_reader`
```

**Comment by wcampbell0x2a** [comment]
Binrw has [NoSeek](https://docs.rs/binrw/latest/binrw/io/struct.NoSeek.html) for this, I'd happy approve a similar solution. 

[Previous.](https://github.com/sharksforarms/deku/pull/360#issuecomment-2327811694) [Conversation.](https://github.com/sharksforarms/deku/pull/360#issuecomment-2328794583)

**Reply by wcampbell0x2a** [reply, marked as answer]
This now exists on master: https://github.com/sharksforarms/deku/pull/487
